packages/hash/integration/src/tests/model/knowledge/user.model.test.ts

```diff
@@ -55,4 +72,14 @@ describe("User model class", () => {
 
     expect(fetchedUser!.entityId).toBe(createdUser.entityId);
   });
+
+  it("can get a user by its kratos identity id", async () => {
+    const fetchedUser = await UserModel.getUserByKratosIdentityId(graphApi, {
+      kratosIdentityId,
+    });
+
+    expect(fetchedUser).not.toBeNull();
+
+    expect(fetchedUser!.entityId).toBe(createdUser.entityId);
```

Comment:
nit:
`toEqual` instead of `toBe`? `toEqual` is (deeply) structual whereas `toEqual` is referential
```suggestion
    expect(fetchedUser!.entityId).toEqual(createdUser.entityId);
```

Reply:
Perhaps I've misunderstood, but I think it's the other way around? ([see the documentation of `toBe`](https://jestjs.io/docs/expect#tobevalue))

Reply:
Just checking the `entityId` was a bit lazy of me though, have changed it to check whether the instances match (which also helped me catch a bug 🎉):
- fixed the bug in https://github.com/hashintel/hash/pull/934/commits/84dc3dab071f6388bdd36055746f2b6cc3bee3cf
- improved the tests in https://github.com/hashintel/hash/pull/934/commits/231432af0630dfb307cea461e1e8ea21cc56430e